src/Api/Endpoints/ImportNotifications/EndpointRouteBuilderExtensions.cs

```diff
@@ -38,14 +39,19 @@ public static void MapImportNotificationsEndpoints(this IEndpointRouteBuilder ap
 
     /// <param name="request">Request</param>
     /// <param name="btmsService">BTMS Service</param>
+    /// <param name="httpContext">HTTP Context</param>
     /// <param name="cancellationToken">Cancellation Token</param>
     [HttpGet]
     private static async Task<IResult> GetUpdated(
         [AsParameters] UpdatedImportNotificationRequest request,
         [FromServices] IBtmsService btmsService,
+        HttpContext httpContext,
         CancellationToken cancellationToken
     )
     {
+        if (!BcpAccessAuthorisation.ClientHasAccessTo(httpContext.User, request.Bcp.ToList()))
```

Comment:
Perhaps in the ticket when we implement JWT integration, this logic could be moved to a filter. But the get endpoint would also need similar and that checks data we will be returning in the response so it's a bit harder.